egs/VAD/binarize.py

```diff
@@ -0,0 +1,72 @@
+from itertools import groupby
+import torch
+
+
+class Binarize(torch.nn.Module):
+    def __init__(self, threshold=0.5, stability=0.1, sample_rate=8000):
+        super().__init__()
+        self.threshold = threshold
+        self.stability = stability
+        self.sample_rate = sample_rate
+
+    def forward(self, x):
+        active = x > self.threshold
+        active = active.squeeze(1).tolist()
+        pairs = count_same_pair(active)
+        active = fit(pairs, self.stability, self.sample_rate)
+        return active
+
+
+def count_same_pair(nums):
+    result = []
+    for num in nums:
+        result.append([[i, sum(1 for _ in group)] for i, group in groupby(num)])
+    return result
+
+
+def fit(pairs, stability, sample_rate):
```

Comment:
Same here.

Maybe you can factor out some of the code into smaller functions (eg. some of the loops' bodies?)